configure

```diff
@@ -96,8 +96,10 @@ check_toolchain()
     if ! command -v $EMACS >/dev/null 2>&1; then
         EMACS=""
     else
-        emacs_version=$($EMACS --version 2>/dev/null | head -n 1)
-        if echo $emacs_version | grep -Eq 'GNU Emacs 2[6789]'; then
+        # extract major version from "GNU Emacs [major].[minor]"
+        emacs_version=$($EMACS --version 2>/dev/null | head -n 1 | cut -d " " -f 3)
+        emacs_major=$(echo "$emacs_version" | cut -d "." -f 1)
```

Comment:
You can get Emacs major version by printing `emacs-major-version`.

```console
> emacs -Q --batch --eval='(message "%s" emacs-major-version)'
30
```

Reply:
Applied that, and also cleaned up the errors a bit.